cartographer/mapping/sparse_pose_graph.h

```diff
@@ -33,9 +34,16 @@ namespace mapping {
 proto::SparsePoseGraphOptions CreateSparsePoseGraphOptions(
     common::LuaParameterDictionary* const parameter_dictionary);
 
-// Splits TrajectoryNodes by ID.
-std::vector<std::vector<TrajectoryNode>> SplitTrajectoryNodes(
-    const std::vector<TrajectoryNode>& trajectory_nodes);
+// TrajectoryNodes and Constraits are provided in a flat vector, but
```

Comment:
typo: `Constraints`. Or better: Remove `and Constraints` since they remain to be in a flat vector.